### PR TITLE
Move Java Lambda builder over to SAM build

### PR DESCRIPTION
**Pull request by abrooksv**
* Re-do all the tests to be Maven/Gradle based

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
Migrates Java LambdaBuilder over to sam build command

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt [hunk 1]
```diff
@@ -82,6 +82,10 @@ abstract class LambdaBuilder {
                 override fun onTextAvailable(event: ProcessEvent, outputType: Key<*>) {
                     // TODO: We should find a way to show the output of this in the UI
                     LOG.info { event.text }
+
+                    if (ApplicationManager.getApplication().isUnitTestMode) {
+                        println("SAM CLI: ${event.text}")
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt]: do we want to keep this?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilder.kt [hunk 2]
```diff
@@ -61,73 +24,19 @@ class JavaLambdaBuilder : LambdaBuilder() {
...
-        if (source.isDirectory) {
-            FileUtil.copyDir(source, dest, false)
-        } else {
-            FileUtil.copy(source, File(dest, source.name))
+    private fun getBaseDirectory(module: Module): VirtualFile {
+        val rootManager = ModuleRootManager.getInstance(module)
+        val contentRoots = rootManager.contentRoots
+        return when (contentRoots.size) {
+            1 -> contentRoots[0]
+            else -> throw IllegalArgumentException("Module can only have 1 content root")
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilder.kt]: Do we need to provide messaging to customers, or is multi-content root rare enough?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilder.kt]: It is currently the only way to debug the unit tests. When we switch over to a pre task we can see if we can remove this.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilder.kt (hunk 2 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilder.kt]: It should be rare in Java land, but ill add a message

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilder.kt]: Added

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilder.kt [hunk 3]
```diff
@@ -61,73 +25,19 @@ class JavaLambdaBuilder : LambdaBuilder() {
...
-
-    private fun copyClasses(module: Module, buildDir: File) {
-        productionRuntimeEntries(module)
-            .withoutLibraries()
-            .classes()
-            .pathsList.pathList
-            .map { File(it) }
-            .filter { it.exists() }
-            .forEach { copyFileOrDir(it, buildDir) }
+        return buildLambdaFromTemplate(module, customTemplate.toPath(), "Function", samOptions)
```
Comment (zhangzhx) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilder.kt]: Do we need to return this magical logical id `Function` from `writeDummySamTemplate` method call? It seems a bit inexplicable here.

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilder.kt]: We should probably just pass it in, its hard coded right now

**Pull request merged**